test/library/draft/arrow/Arrow.chpl

```diff
@@ -0,0 +1,253 @@
+module Arrow {
+    require "ArrowDecl.chpl";
+    public use ArrowDecl;
+    public use SysCTypes, CPtr;
+    // -------------------------- Type Declarations and Functions -------------
+
+    type ArrowArray = GArrowArray;
+
+    proc arrowInt32(): GArrowInt32DataType {
+        return garrow_int32_data_type_new();
+    }
+    
+    proc arrowInt64(): GArrowInt32DataType {
+        return garrow_int64_data_type_new();
+    }
+    proc arrowString(): GArrowStringDataType {
+        return garrow_string_data_type_new();
+    }
+    proc arrowBool(): GArrowBooleanDataType {
+        return garrow_boolean_data_type_new();
+    }
+    //--------------------------- Array building functions --------------------
+
+    proc array(array: [?arrDom] ?arrayType, validity: [?validityDom] int = [-1], 
+                specifiedType = nil): GArrowArray {
+        var fakeReturn: GArrowArray;
+        // Sorting based on the specifiedType and calling the appropriate function
+        //if(!specifiedType)
+        select arrayType {
+            when int do return int64Array(array, validity);
+            when string do return stringArray(array, validity);
+            when bool do return boolArray(array, validity);
+            otherwise {
+                writeln ("Unsupported type, \nreturning nil"); 
+                return fakeReturn;
+            }
+        }
+    }
+    
+    proc int64Array(array: [?arrDom] int, validity: [?validityDom] int) : GArrowInt64Array {
+        
+        var retval: GArrowInt64Array;
+        
+        var builder: GArrowInt64ArrayBuilder;
+        var success: gboolean = 1;
+        var error: GErrorPtr;
+
+        builder = garrow_int64_array_builder_new();
+        //writeln("0");
+
+        if (success) {
+            //writeln("1");
+            var intArrLen: gint64 = array.size: gint64;
+            //writeln("1.1 + ", intArrLen, " " , array.size);
+            var intValArr: [0..#intArrLen] gint64;
+            forall i in 0..#intArrLen {
+                intValArr[i] = array[i]: gint64;
+            }
+            //writeln("2 + ", intValArr);
+
+            var intValidityArr: [0..#intArrLen] gboolean;
+            forall i in 0..#intValidityArr.size {
+                intValidityArr[i] = 1;
+            }
+            if(validity[0] != -1){
+                forall i in 0..#validity.size {
+                    intValidityArr[validity[i]] = 0;
+                }
+            }
```

Comment:
Same as above, these can be condensed into one line assign statements (can't remember the real name). If this is only place that we are using that -1 in the validity array, as I was bringing up earlier, this would be easy to have it just start as an empty array instead.